OSDOCS-4652: adding Additional information assembly and mods to Configuration (#53516)
openshift/openshift-docs: modules/microshift-certificate-lifetime.adoc

// Module included in the following assemblies:
//
// * microshift/microshift-things-to-know.adoc

:_content-type: CONCEPT
[id="microshift-certificate-lifetime_{context}"]
= Security certificate lifetime
{product-title} certificates are separated into two basic groups:

. Short-lived certificates having certificate validity of one year.
. Long-lived certificates having certificate validity of 10 years.

Most server or leaf certificates are short-lived.

An example of a long-lived certificate is the client certificate for `system:admin user` authentication, or the certificate of the signer of the `kube-apiserver` external serving certificate.

[id="microshift-certificate-rotation_{context}"]
== Certificate rotation

> As certificates age, {product-title} can be restarted to rotate certificates. A certificate that is close to expiring may also automatically cause a restart. Read the following situation overviews to understand the actions at each moment in time:

Reviewer: Use `might instead of `may`

. Green zone:
.. When a short-term certificate is 5 months old, no rotation occurs.
.. When a long-term certificate is 8.5 years old, no rotation occurs.

. Yellow zone:
.. When a short-term certificate is 8 months old, it is rotated when {product-title} starts or restarts.
.. When a long-term certificate is 9 years old, it is rotated when {product-title} starts or restarts.

. Red zone
.. When a short-term certificate is 8 months old, {product-title} restarts to rotate and apply a new certificate.
.. When a long-term certificate is 9 years old, {product-title} restarts to rotate and apply a new certificate.

[NOTE]
====

> If the rotated certificate is a Certificate Authority, all of the certificates it signed rotate.

Reviewer: ```suggestion
If the rotated certificate is a Certificate Authority, the certificates that it signed rotate.
```

====

.Stop-light timeline of {product-title} certificate validity.

> image::microshift-cert-rotation.png[<{product-title} graph with symbolic green-yellow-red stoplight map of certificates>]

Reviewer: The section name says `Stop-light`. Can you check if it has to be stop-light or stoplight?

Author: IMB does not use the hypen. Good catch, thanks1
